### PR TITLE
Language support

### PR DESCRIPTION
**Pull request by ilonamm**
#38 

I have set up gettext, but still need to:

- [x] add gettext to all strings on landing page

- [ ] add gettext to all strings on share page

- [x] automatically guess the locale

Also I created two translation files, fi and pt-BR

At the moment can access them like this:
 http://localhost:4000/?locale=pt-BR

Just putting this here to test in QA but not really ready for production yet.

### CODE REVIEW

**Comment by ilonamm** [comment]
Full landing page has gettext tags, and we have 1st version of full Finnish translation.

Issues:
- [ ] Resolve this warning, not sure from which dependency
[error] The client's requested channel transport version "2.0.0" does not match server's version requirements of "~> 1.0"

- [ ] nominate button and placeholder for autoselect are missing gettext notation, therefore not translated

**Comment by ilonamm** [comment]
Additional todo:

- [x] Make a language navigation menu to allow user change between supported languages

- [ ] Add locale in the db so that we know which language the user prefers, for example, for emails

@batmany13 could this error be because of phoenix 1.2? should we migrate to 1.3?
`[error] The client's requested channel transport version "2.0.0" does not match server's version requirements of "~> 1.0"`

**Comment by ilonamm** [comment]
One more todo.. 

- [x] Fix this CaseClauseError

Aug 13 11:33:21 askbywho-qa app/web.1: Request: GET /?locale=pt-BR 
Aug 13 11:33:21 askbywho-qa app/web.1: ** (exit) an exception was raised: 
Aug 13 11:33:21 askbywho-qa app/web.1:     ** (CaseClauseError) no case clause matching: [] 
Aug 13 11:33:21 askbywho-qa app/web.1:         (askbywho) web/controllers/locale.ex:15: Askbywho.Locale.call/2

**Comment by ilonamm** [comment]
@ohanhi @jpsb @lucasmncastro @lucasarlop 
Hi JS experts, could you help with this:
Made a function `add_autocomplete_placeholder()` in `page_view.ex`, need to make the `pages.js` row 34 fetch the text from there to allow it to access translations. I'm total JS newbie, what do I need here? :D 
`placeholder: 'Click to write/choose brands'`

File: web/static/css/home.css [hunk 1]
```diff
@@ -270,6 +280,52 @@ h1 {
...
+  text-align: center;
+  justify-content: center;
+  font-weight: 600;
+}
+.about-askbywho__share img {
+  width: 52px;
+  padding-right: 5px;
+}
+.about-askbywho__share a {
+   text-decoration: none;
```
Comment (ilonamm) [inline web/static/css/home.css:294]: Having an issue with a very persistent link text-decoration... @ohanhi could you give a look?

**Comment by ilonamm** [comment]
Added footer (#33)

**Review by batmany13 — COMMENTED** [review]
why are the "priv/.DS_STORE" files checked in?

File: priv/gettext/pt-BR/LC_MESSAGES/default.po [hunk 2]
```diff
@@ -0,0 +1,196 @@
...
+
+#: web/templates/page/index.html.eex:178
+msgid "Do you want to wear clothes made in decent working conditions?"
+msgstr ""
+
+#: web/templates/page/index.html.eex:90 web/templates/page/index.html.eex:92
+#: web/templates/page/index.html.eex:94 web/templates/page/index.html.eex:97
+#: web/templates/page/index.html.eex:99 web/templates/page/index.html.eex:101
+msgid "I don't want to buy clothes made by"
+msgstr ""
```
Comment (batmany13) [inline priv/gettext/pt-BR/LC_MESSAGES/default.po:100]: are these just pending translations?

File: web/controllers/locale.ex [hunk 3]
```diff
@@ -0,0 +1,50 @@
...
+    "fi-fi" => "fi"}
+
+    # TODO add when portuguese translation is ready
+    # "pt-br" => "pt-BR", "pt" => "pt-BR"
+
+    # fetching browser's guess of what languages the user prefers
+    case get_req_header(conn, "accept-language") do
+      [language] ->
+        pattern  = :binary.compile_pattern([",", ";"])
+        IO.puts("accept-language is #{language}")
```
Comment (batmany13) [inline web/controllers/locale.ex:19]: use logger

**Review by batmany13 — APPROVED** [review]
great work on the i18n work.  Truly an international product

File: priv/gettext/pt-BR/LC_MESSAGES/default.po (hunk 2 above)
Comment (ilonamm) [reply priv/gettext/pt-BR/LC_MESSAGES/default.po:100]: yes, file set up for portuguese translation!

**Pull request merged**